tests/addEmailIdentityWorkflow.test.tsx

```diff
@@ -0,0 +1,77 @@
+import { enableFetchMocks } from "jest-fetch-mock";
+import React from "react";
+
+import { IdentityTypes } from "../src/@types/Identity";
+import { ProviderUser } from "../src/@types/ProviderUser";
+import Index from "../src/pages/index";
+import { render, fireEvent } from "./config/test-utils";
+
+enableFetchMocks();
+
+describe("addEmailIdentityWorkflow", () => {
+  type MockedFetchedData = {
+    data: {
+      provider: ProviderUser;
+    };
+  };
+
+  const mockedFetchedData: MockedFetchedData = {
+    data: {
+      provider: {
+        id: "4a5f8589-0d91-4a69-924a-6f227a69666d",
+        name: "Mocked Provider",
+        user: {
+          id: "299d268e-3e19-4486-9be7-29c539d241ac",
+          identities: [
+            {
+              id: "7f8d168a-3f65-4636-9acb-7720a212680e",
+              primary: true,
+              status: "validated",
+              type: IdentityTypes.PHONE,
+              value: "0123456789",
+            },
+            {
+              id: "8f79dcc1-530b-4982-878d-33f0def6a7cf",
+              primary: true,
+              status: "validated",
+              type: IdentityTypes.EMAIL,
+              value: "test@test.test",
+            },
+          ],
+        },
+      },
+    },
+  };
+
+  test("Add email identity user flow with button click", async () => {
+    const { getByText, getByRole, getByTestId } = render(
+      <Index fetchedData={mockedFetchedData} />,
+      {},
+    );
+
+    const addAnEmailButton = getByText("Add an email");
+
+    fireEvent.click(addAnEmailButton);
+
+    const identityForm = getByTestId("identity-form");
+
+    let submitCheck = false;
+
+    const eventListenerCallback = (event: Event): void => {
+      submitCheck = true;
+      event.currentTarget?.removeEventListener("submit", eventListenerCallback);
+    };
+
+    identityForm.addEventListener("submit", eventListenerCallback);
+
+    const emailInput = getByRole("textbox");
+
+    fireEvent.change(emailInput, { target: { value: "test@test.test" } });
+
+    const sendEmailButton = getByText("Send");
+
+    fireEvent.click(sendEmailButton);
+
+    expect(submitCheck).toBe(true);
```

Comment:
I have a little fear that this could be a flacky test in the future since we trigger a click that would trigger an event that could be asynchronous. Are you sure it always work?

Reply:
I am not, I'll spend more time on this playing with async processes.

Reply:
I tested with adding an async api call on `https://api.spacexdata.com/v4/launches/latest` on submit, and logging its result. Everything works fine.

Worst case, we would need to use [their async helper](https://testing-library.com/docs/dom-testing-library/api-async) if this stop working. 

Let me know if you think it's good enough or not.

Reply:
As seen together, it is working with async actions.